src/main/java/rodain/source/representation/SourceDirectory.java

```diff
@@ -57,41 +60,46 @@ public void closeDirectoryStream(){
             if(directoryStream != null)
                 directoryStream.close();
         } catch (IOException e) {
-            log.error(e.getMessage());
+            log.error("" + e);
         }
     }
 
     private void startDirectoryStream(){
-        if(directoryStream != null) return;
+        if(directoryStream != null)
+            return;
 
         try {
             directoryStream = Files.newDirectoryStream(path);
             iterator = directoryStream.iterator();
         }
         catch (AccessDeniedException e){
-            log.info(e.getMessage());
+            log.info("" + e);
         }
         catch (IOException e) {
-            log.error(e.getMessage());
+            log.error("" + e);
         }
     }
 
-    public boolean hasFirstLoaded(){return children.size() > 1;}
+    public boolean hasFirstLoaded(){
+        return children.size() > 1;
+    }
 
-    public TreeMap<String, SourceItem> loadMore(){
+    public SortedMap<String, SourceItem> loadMore(){
         startDirectoryStream();
         int loaded = 0, childrenSize = children.size();
-        TreeMap<String, SourceItem> result = new TreeMap<String, SourceItem>();
+        TreeMap<String, SourceItem> result = new TreeMap<>();
         itemsToLoad += 50;
```

Comment:
The number 50 here should be a constant, and maybe even a configuration on a properties file.